web/src/compose_actions.js

```diff
@@ -70,6 +71,35 @@ function show_compose_box(msg_type, opts) {
     // When changing this, edit the 42px in _maybe_autoscroll
     $(".new_message_textarea").css("min-height", "3em");
     compose_ui.set_focus(msg_type, opts);
+    const filter = narrow_state.filter();
+    if (filter && !filter.is_common_narrow()) {
+        show_compose_in_search_view_banner();
+    }
+}
+
+function show_compose_in_search_view_banner() {
+    // TODO: use compose_state.stream_id() when #25785 merges.
+    let stream_id = "";
+    const stream_name = compose_state.stream_name();
+    if (stream_name) {
```

Comment:
We shouldn't show `Go to conversation` when `stream_name` is empty.

Reply:
In that case, should we have a banner without a button which would only disappear once the narrow is exited?

Reply:
Is there a reason we are doing this only for streams and not DMs in search results?

Reply:
No, there isn't. I'll update to include DMs.